crates/solver/src/settlement_submission/submitter.rs

```diff
@@ -421,6 +429,20 @@ impl<'a> Submitter<'a> {
     }
 }
 
+fn status(receipt: TransactionReceipt) -> Result<TransactionReceipt, SubmissionError> {
+    if let Some(status) = receipt.status {
+        if status == U64::zero() {
+            // failing transaction
+            return Err(SubmissionError::Revert(None));
```

Comment:
As a future improvement we could try and detect the revert message by re-simulating the transaction at the same block and transaction index. Not needed for this PR though.